symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/SessionService.java

```diff
@@ -25,10 +28,7 @@
    * @return Bot session info.
    */
   public UserV2 getSession(AuthSession authSession) {
-    try {
-      return this.sessionApi.v2SessioninfoGet(authSession.getSessionToken()) ;
-    } catch (ApiException ex) {
-      throw new ApiRuntimeException(ex);
-    }
+    return RetryWithRecovery.executeAndRetry("getSession",
```

Comment:
Passing the `BdkRetryConfig` is a bit repetitive, could you modify your design to avoid this? 

I remember that we talked about a `RetryFactory` or equivalent that would have been initialized at `SymphonyBdk` level, including configuration

Reply:
Yes but we still have to pass something to the classes, either a `BdkRetryConfig` or a `RetryWithRecoveryBuilder` initialised with the config. In either cases, we have to pass something to every classes.
Or maybe I didn't get something in your proposal.

Reply:
Correct, we would still have to pass something to services. But IMHO the code would be more functional if we pass "a retry thing" rather than "a retry config"